presto-main/src/main/java/com/facebook/presto/sql/planner/plan/RowNumberNode.java

```diff
@@ -107,4 +107,9 @@ public Symbol getRowNumberSymbol()
     {
         return visitor.visitRowNumber(this, context);
     }
+
+    public RowNumberNode withLimit(int maxRowCountPerPartition)
+    {
+        return new RowNumberNode(getId(), source, partitionBy, rowNumberSymbol, Optional.of(maxRowCountPerPartition), hashSymbol);
```

Comment:
if RowNumberNode already has maxRowCountPerPartition, maybe you should apply the min of the two?